src/files/mv.js

```diff
@@ -3,12 +3,24 @@
 const promisify = require('promisify-es6')
 
 module.exports = (send) => {
-  return promisify((args, opts, callback) => {
-    if (typeof opts === 'function' &&
-      callback === undefined) {
-      callback = opts
-      opts = {}
+  return promisify(function () {
+    const args = Array.prototype.slice.call(arguments)
+    const callback = args.pop()
+    let opts = {}
+    let sources = []
+
+    if (!Array.isArray(args[args.length - 1]) && typeof args[args.length - 1] === 'object') {
+      opts = args.pop()
+    }
+
+    if (args.length === 1 && Array.isArray(args[0])) {
+      // support ipfs.file.mv([src, dest], opts, cb)
+      sources = args[0]
+    } else {
+      // support ipfs.file.mv(src, dest, opts, cb) and ipfs.file.cp(src1, src2, dest, opts, cb)
+      sources = args
     }
```

Comment:
Maybe we can have the above logic abstracted in a util function since the code is replicated in `mv` and `cp`. What do you think @achingbrain ?